web/pgadmin/tools/sqleditor/static/js/components/sections/StatusBar.jsx

```diff
@@ -50,6 +51,35 @@ export function StatusBar() {
   });
   const {seconds, minutes, hours, msec, start:startTimer, pause:pauseTimer, reset:resetTimer} = useStopwatch({});
 
+  const [selectedEol, setSelectedEol] = useState(OS_EOL);
+  const [openMenuName, setOpenMenuName] = useState(null);
+  const prevMenuOpenIdRef = React.useRef(null);
+  const eolMenuRef = React.useRef(null);
+
+  const toggleMenu = React.useCallback((e) => {
+    const name = e.currentTarget?.getAttribute('name');
+    setOpenMenuName(() => {
+      return prevMenuOpenIdRef.current === name ? null : name;
+    });
+    prevMenuOpenIdRef.current = null;
+  }, []);
+
+  const onMenuClose = React.useCallback(() => {
+    prevMenuOpenIdRef.current = openMenuName;
+    setTimeout(() => {
```

Comment:
what is this for? Why are you not using `usePgMenuGroup`?

Reply:
Actually i took reference from `ResultSetToolbar.jsx` ,
Will update with `usePgMenuGroup`.

Reply:
Then change there as well.